sdks/java/extensions/sql/zetasql/src/main/java/org/apache/beam/sdk/extensions/sql/zetasql/BeamZetaSqlCatalog.java

```diff
@@ -379,6 +391,16 @@ private void addUdfsFromSchema() {
     }
   }
 
+  private List<FunctionArgumentType> getArgumentTypes(
```

Comment:
nit: Looks like you could just move this to before the `if` block like you did with path rather than making it a function? I have no strong opinion here but that might make this a bit simpler.

Reply:
done

Reply:
Never mind, the order did matter here, since we need to validate scalar parameter types before converting them.